pkg/controller/registry/grpc/source.go

```diff
@@ -163,7 +163,7 @@ func grpcConnection(address string) (*grpc.ClientConn, error) {
 		}))
 	}
 
-	return grpc.Dial(address, dialOptions...)
+	return grpc.NewClient(address, dialOptions...)
```

Comment:
I'm not sure we should make this change. I see that `grpc.Dial` is deprecated, but reading the GoDoc of `grpc.DialContext` (which is called by `grpc.Dial`), it sounds like there are subtle differences that may matter to us:

https://github.com/grpc/grpc-go/blob/c52d02553f2649c28d0279bca17909a3a08de022/clientconn.go#L228-L236

> // One subtle difference between NewClient and Dial and DialContext is that the
> // former uses "dns" as the default name resolver, while the latter use
> // "passthrough" for backward compatibility.  **This distinction should not matter
> // to most users, but could matter to legacy users that specify a custom dialer
> // and expect it to receive the target string directly.**

We _do_ use a custom dialer if there is a proxyURL.